sdk/main/src/Transaction.cc

```diff
@@ -788,6 +817,41 @@ void Transaction<SdkRequestType>::buildAllTransactions() const
   }
 }
 
+template<typename SdkRequestType>
+void Transaction<SdkRequestType>::buildAllUnsignedTransactions()
+{
+  // clearing mTransactions as every time toBytes would call this we want to
+  // have a clean repopulate of the transactions
+  mImpl->mTransactions.clear();
+  // build body
+  updateSourceTransactionBody(nullptr);
+  // populate mTransactions
+
+  // As specified in HIP-745 we can create dummy nodeAccountIds they can be later
+  // repopulated after deserialization. The need for this dummy value stands
+  // because otherwise the Transaction will be executed on client network by
+  // default from past specifications
+  if (Executable<SdkRequestType, proto::Transaction, proto::TransactionResponse, TransactionResponse>::
+        getNodeAccountIds()
+          .empty())
+  {
+    setNodeAccountIds({ AccountId() });
+  }
+  // If nodeAccountIds are set we should try to send the request to them
+  for (const AccountId& accountId :
+       Executable<SdkRequestType, proto::Transaction, proto::TransactionResponse, TransactionResponse>::
+         getNodeAccountIds())
+  {
+    // Since transaction proto bodybytes are deprecated we are working around with
+    // the use of signedTransaction proto bodybytes
+    mImpl->mSourceTransactionBody.set_allocated_nodeaccountid(accountId.toProtobuf().release());
+    proto::SignedTransaction signedTransaction;
+    signedTransaction.set_bodybytes(mImpl->mSourceTransactionBody.SerializeAsString());
+    // populating mTransactions which should be further serialized
+    addTransaction(signedTransaction);
+  }
```

Comment:
This `for`-loop looks like the same processing that's contained in `addSignedTransactionForEachNode()`, could it be replaced with that?

Reply:
It has some differences so I decided not to reuse the function, because that would mean reworking it.